internal/pgbackrest/config.go

```diff
@@ -263,6 +263,67 @@ mv "${pgdata}" "${pgdata}_bootstrap"`
 	return append([]string{"bash", "-ceu", "--", restoreScript, "-", pgdata}, args...)
 }
 
+// DedicatedSnapshotVolumeRestoreCommand returns the command for performing a pgBackRest delta restore
+// into a dedicated snapshot volume. In addition to calling the pgBackRest restore command with any
+// pgBackRest options provided, the script also removes the patroni.dynamic.json file if present. This
+// ensures the configuration from the cluster being restored from is not utilized when bootstrapping a
+// new cluster, and the configuration for the new cluster is utilized instead.
+func DedicatedSnapshotVolumeRestoreCommand(pgdata string, tablespaceVolumes []*corev1.PersistentVolumeClaim, args ...string) []string {
+
+	// The postmaster.pid file is removed, if it exists, before attempting a restore.
+	// This allows the restore to be tried more than once without the causing an
+	// error due to the presence of the file in subsequent attempts.
+
+	tablespaceCmd := ""
+	for _, tablespaceVolume := range tablespaceVolumes {
+		tablespaceCmd = tablespaceCmd + fmt.Sprintf(
+			"\ninstall --directory --mode=0700 '/tablespaces/%s/data'",
+			tablespaceVolume.Labels[naming.LabelData])
+	}
+
+	// Wrap pgbackrest restore command in backup_label checks. If pre/post
+	// backup_labels are different, restore moved database forward, so return 0
+	// so that the Job is successful and we know to proceed with snapshot.
+	// Otherwise return 1, Job will fail, and we will not proceed with snapshot.
+	restoreScript := `declare -r pgdata="$1" opts="$2"
+BACKUP_LABEL="xxx"
+if [[ -e "` + pgdata + `/backup_label" ]]
+then
+  if BACKUP_LABEL=$(md5sum ` + pgdata + `/backup_label); then
+    BACKUP_LABEL=$(echo "${BACKUP_LABEL}" | awk '{print $1}')
+  else
+    echo "Error getting checksum of backup_label."
+    exit 1
+  fi
+fi
```

Comment:
⛏️ It looks like `pgdata` (Go) is passed into the script and assigned to `pgdata` (Bash). If so, these string concat can be replaced with shell strings:

```diff
-if [[ -e "` + pgdata + `/backup_label" ]]
+if [[ -e "${pgdata}/backup_label" ]]
```

:pick: The label command looks the same both before and after. If so, we can drop the call to `awk`.

:memo: `bash -e` halts the script whenever a command list exits non-zero.

🤔 If the error message from `md5sum` is sufficient, we can combine all that and let it halt the script:

```suggestion
	restoreScript := `declare -r pgdata="$1" opts="$2"
BACKUP_LABEL=$([[ ! -e "${pgdata}/backup_label" ]] || md5sum "${pgdata}/backup_label")
```